modules/sdk-coin-sol/src/lib/transactionBuilder.ts

```diff
@@ -76,6 +76,15 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     this.feePayer(txData.feePayer as string);
     this.nonce(txData.nonce, txData.durableNonce);
     this._instructionsData = instructionParamsFactory(tx.type, tx.solTransaction.instructions);
+    // Set priority fee if it exists
+    const filteredPriorityFeeInstructionsData = txData.instructionsData.filter(
+      (data) => data.type === InstructionBuilderTypes.SetPriorityFee
+    );
+
+    if (filteredPriorityFeeInstructionsData.length > 0) {
+      const priorityFeeInstructionsData = filteredPriorityFeeInstructionsData[0];
+      this._priorityFee = Number(priorityFeeInstructionsData.params.fee);
+    }
```

Comment:
```
if (instruction.type === InstructionBuilderTypes.SetPriorityFee) {
        this.setPriorityFee()
      }
```

We are already iterating over the instructions in below for loop, lets add this condition there itself. Also we can use the setter instead of setting the value directly